src/util-memcmp.c

```diff
@@ -156,154 +156,427 @@ static int MemcmpTest13 (void)
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    "4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa4096aaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+    NULL,
+};
+
+static inline int TestWrapMemcmp(const void *s1, const void *s2, size_t len)
+{
+    return (memcmp(s1, s2, len) == 0) ? 0 : 1;
+}
 
-    printf("r1 %d\n", r1);
-    FAIL_IF(r1 != (51 * TEST_RUNS));
+#define BIGSZ 1024 * 1024
+#define DRIVER(f)                                                                                  \
```

Comment:
Could we get this as an inline function?

Reply:
yeah just in the test, so leaving as is